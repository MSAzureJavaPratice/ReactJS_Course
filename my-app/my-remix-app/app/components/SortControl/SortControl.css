.sort-control {
    margin-bottom: 20px;
}

.sort-control select {
    margin-top: 1rem;
    background-color: red;
}