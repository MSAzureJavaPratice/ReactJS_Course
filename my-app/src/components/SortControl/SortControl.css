.sort-control {
  margin-bottom: 20px;
}
