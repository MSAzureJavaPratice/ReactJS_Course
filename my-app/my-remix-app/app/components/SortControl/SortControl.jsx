import React from 'react';
import "./SortControl.css";

const SortControl = ({ currentSort, searchParams }) => {
    // Retrieve all current search params, exclude 'sortBy'
    const currentParams = Array.from(searchParams.entries())
      .filter(([key]) => key !== 'sortBy');

    return (
        <div className="sort-control">
            <form action="/" method="GET">
                <label htmlFor="sort-by">Sort by:</label>
                
                    <select id="sort-by" name="sortBy" defaultValue={currentSort}>
                        <option value="release_date">Release Date</option>
                        <option value="title">Title</option>
                    </select>
                    {/* Propagate other parameters as hidden inputs */}
                    {currentParams.map(([key, value]) => (
                        <input key={key} type="hidden" name={key} value={value} />
                    ))}
                    {/* Submit button visually optimized to appear integrated */}
                    <input type="submit" value="" style={{ padding: "0px", background: "red", position: "absolute", marginTop: "25px", marginLeft: "-12px", height: "0px" }} />
            </form>
        </div>
    );
};

export default SortControl;