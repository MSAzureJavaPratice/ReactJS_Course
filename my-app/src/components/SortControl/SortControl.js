import React from "react";
import "./SortControl.css";

// SortControl component
const SortControl = ({ currentSort, onSortChange }) => {
  const handleSortChange = (event) => {
    onSortChange(event.target.value);
  };

  return (
    <div className="sort-control">
      <label htmlFor="sort-by"> Sort by: </label>{" "}
      <select id="sort-by" value={currentSort} onChange={handleSortChange}>
        <option value="releaseDate"> Release Date </option>{" "}
        <option value="title"> Title </option>{" "}
      </select>{" "}
    </div>
  );
};

export default SortControl;
