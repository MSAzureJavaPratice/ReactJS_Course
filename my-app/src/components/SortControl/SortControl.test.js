import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import SortControl from "./SortControl";

describe("SortControl Component", () => {
  const onSortChange = jest.fn();

  test("renders sort control correctly", () => {
    render(
      <SortControl currentSort="releaseDate" onSortChange={onSortChange} />
    );
    expect(screen.getByLabelText("Sort by:")).toBeInTheDocument();
    expect(screen.getByDisplayValue("Release Date")).toBeInTheDocument();
  });

  test("calls onSortChange when selecting a new option", () => {
    render(
      <SortControl currentSort="releaseDate" onSortChange={onSortChange} />
    );
    fireEvent.change(screen.getByLabelText("Sort by:"), {
      target: { value: "title" },
    });
    expect(onSortChange).toHaveBeenCalledWith("title");
  });

  test("matches snapshot", () => {
    const { asFragment } = render(
      <SortControl currentSort="releaseDate" onSortChange={onSortChange} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
