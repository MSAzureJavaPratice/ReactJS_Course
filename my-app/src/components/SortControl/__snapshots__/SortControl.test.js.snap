// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortControl Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sort-control"
  >
    <label
      for="sort-by"
    >
       Sort by: 
    </label>
     
    <select
      id="sort-by"
    >
      <option
        value="releaseDate"
      >
         Release Date 
      </option>
       
      <option
        value="title"
      >
         Title 
      </option>
       
    </select>
     
  </div>
</DocumentFragment>
`;
